### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by helciodev**
This pull request adds the user interface and the general game flow

### CODE REVIEW

**Comment by mimipeshy** [comment]
### STATUS : CHANGES REQUESTED :red_circle:  :recycle: 
Hello @helciodev  and @kunmi02 :wave:  :wave: , good job on working on this project, it looks good and works but there are some things you have left out noted below :arrow_down: 

**Note**: _usage of inline comment is OK in this milestone_

##### Behavior
- In this step students are supposed to only **mock** the basic flow of the game. This means implementing a user interface that displays prompt for users (print/puts) and collects user input (`gets.chomp`). 
- No game logic (for example checking who is a winner) is required in this milestone - instead, students can just display "The winner is X" no matter who won. 
* [x] `[REQUIRED]` game tells which player turn it is (without computing who the winner is)
* [x] `[REQUIRED]` game asks the player to select from available moves
* [x] `[REQUIRED]` game informs player if the selected move is invalid (without computing what an invalid move is)
* [x] `[REQUIRED]` game displays board after the player move (without computing which spots on the board are taken)
* [x] `[REQUIRED]` game informs player if the selected move is a winning move (without computing the winning move)
* [x] `[REQUIRED]` game informs player if the selected move is a draw move (without computing draw)
* [x] `[REQUIRED]` game repeats all actions for the next player's move

##### Implementation details

* [x] A loop is needed in this game (recommended is `while` but anything else will do) ([OdinProject hint](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/loop.png))
* [x] Inside the loop, the student should call actions repeated for each move
* [x] A loop needs to be terminated at some point (after a winning move or draw to be more specific). It can be done with variable:

```ruby
game_on = true

# initial instructions for player

while game_on
  # loop for each move

  if winner # and/or draw (the exact condition in this milestone is not important)
    game_on = false
  end
end
```

That's all :smile: , kindly work on the feedback and submit another review when done.

Regards
Peris Ndanu
Feel free to reach out  to me on slack if you have any questions(perisndanu)

File: bin/main [hunk 1]
```diff
@@ -1,3 +1,118 @@
...
+    check_empty_cell(1, 2)
+  when '7'
+    check_empty_cell(2, 0)
+  when '8'
+    check_empty_cell(2, 1)
+  when '9'
+    check_empty_cell(2, 2)
+  else
+    p 'Please enter a valid input between 1-9'
+  end
```
Comment (SpaYco) [inline bin/main]: - [x] game logic is for the next step, please remove this.

**Review by SpaYco — CHANGES_REQUESTED** [review]
# Changes Requested 

Hello Team! 👋 
Awesome job so far!! 👍 you still need to make some changes though! 

## Required Changes ♻️ 



- [ ]  the game informs player if the selected move is a winning move (without computing the winning move)

- [ ] the game informs player if the selected move is a draw move (without computing draw)

- [ ] I'm getting an error `undefined local variable or method `play' for main:Object (NameError)`, please comment out/remove all calls to undefined methods for now, until you get to next step, until then, the program should print everything without breaking, it's okay if it just prints stuff without using logic, this is what this step is for. ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/OOP/milestone2/logic.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/59565018/))

### please make the appropriate changes and ask for another code review ✔️ 

Happy Coding!

Best, 
Aziz Mejri 
_contact me on [twitter](https://twitter.com/iSpaYco) or slack (@SpaYco) if you need any help_

File: bin/main (hunk 1 above)
Comment (kunmi02) [reply bin/main]: The last reviewer asked us to add that to the code. @spayco you guys are making this review complex

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Azeez Azeez & Helcio André** 
Good job with the improvements for this project :muscle:  :muscle:
It meets most of the requirements :clap: :clap: :clap:  
There are just a couple of details to improve before you can go ahead:

- [ ] You should mock a function that checks if the movement was a draw movement.
- [ ] You should mock a function that checks if the movement was a winner movement.
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by peterrobert — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED :recycle: :stop_sign: 

HELLO @helciodev NICE JOB :+1:  YOU HAVE DONE ALMOST EVERY REQUIREMENT JUST A FEW MORE CHANGES TO BE MADE. KINDLY CHECK BELOW :arrow_down_small:  

- [ ] [REQUIRED] game displays board after the player move (without computing which spots on the board are taken) . What this means is that after every move by a player, you should display the board. 

- [ ] `[REQUIRED]` game informs player if the selected move is a winning move (without computing the winning move)

- [ ] `[REQUIRED]` game informs player if the selected move is a draw move (without computing draw)

### if you have any question kindly reach me on slack @peterrobert
### HAPPY CODING :computer:

**Comment by kunmi02** [comment]
@peterrobert can you please shed more light on the 2nd and 3rd requirements. How are we to implement those requirements without adding the game logic? 

These changes are part of the game. The goal of this milestone is to mock the user interface.

**Review by andresporras3423 — CHANGES_REQUESTED** [review]
### Status: Changes Required :recycle:
**Hello Azeez Azeez & Helcio André** 
Good job with the improvements for this project :muscle:  :muscle:
it meets most of the requirements :clap: :clap: :clap:  
Just a couple of details to improve before you can go ahead:

- [ ] add a function to check if names are valid (both names must be different and they cannot be empty).
- [ ] fix rubocop offenses.
 
Please request another review once you are ready! :+1: 
 
Happy coding :relaxed:
 
Oscar Andrés Russi Porras

**Review by voscarmv — APPROVED** [review]
# Project status: Approved ✨ 😎👍 💯 ✔️
### You may merge this branch and move forward ⛙🛣
### Questions or comments? Reach me on [Twitter](https://twitter.com/voscarmv) or [LinkedIn](https://linkedin.com/in/oscar-mier) 🧑‍💻🗨️🧑‍💻 
# Happy Hacking! 🧠 ⌨️ 🖱️ 🖥️

**Pull request merged**